### ISSUE TITLE
Change Battery indication icon behavior & threshold not working

### ISSUE THREAD

**Issue by maniacx** [maniacx/Battery-Health-Charging#105]
@MaximalCats

Can you give some details.

1. Make model of your laptop.
2. Distro name (example Ubuntu 24.04)
3. Kernel version
4. Gnome shell version

**Comment by MaximalCats** [comment]
**Model:** ThinkPad P50
**Distro:** Void Linux
**Kernel:** 6.6.45_1
**GS version:** 46.2

**Comment by maniacx** [comment]
Is this dual battery or single battery?

First let see what the problem with threshold is.
Copy these command using copy button and paste it in terminal. post the ouptut

```
echo '--- polkit version ---'
pkexec --version
echo '--- writing threshold1---'
echo '55' | sudo tee /sys/class/power_supply/BAT0/charge_control_start_threshold
echo '60' | sudo tee /sys/class/power_supply/BAT0/charge_control_end_threshold
echo '--- reading threshold1---'
cat /sys/class/power_supply/BAT0/charge_control_end_threshold
cat /sys/class/power_supply/BAT0/charge_control_start_threshold
echo '--- writing threshold2---'
echo '100' | sudo tee /sys/class/power_supply/BAT0/charge_control_end_threshold
echo '95' | sudo tee /sys/class/power_supply/BAT0/charge_control_start_threshold
echo '--- reading threshold2---'
cat /sys/class/power_supply/BAT0/charge_control_end_threshold
cat /sys/class/power_supply/BAT0/charge_control_start_threshold
echo '--- writing threshold3---'
echo '75' | sudo tee /sys/class/power_supply/BAT0/charge_control_start_threshold
echo '80' | sudo tee /sys/class/power_supply/BAT0/charge_control_end_threshold
echo '--- reading threshold3---'
cat /sys/class/power_supply/BAT0/charge_control_end_threshold
cat /sys/class/power_supply/BAT0/charge_control_start_threshold
```

You might have to change your BAT0 to BAT1 is name assigned to the battery is BAT1
To find out use command
```
ls -l  /sys/class/power_supply/
```

**Comment by MaximalCats** [comment]
Sorry; battery threshold seems to be working fine now. Icon is still behaving incorrectly.

**Comment by maniacx** [comment]
Which icon are you talking about?

1 Extensions Threshold Icon
![1](https://github.com/user-attachments/assets/74a67609-311b-41fe-87be-6022e80e7d3c)

Or

2. Gnome system tray Battery level icon
![2](https://github.com/user-attachments/assets/8d375dcf-cffa-4767-87d1-5d3138036183)

Explain how is it behaving incorrectly?
And explain what is expected behavior?

**Comment by MaximalCats** [comment]
Talking about the sceond one. Icon doesn't change (says charging while at threshold). From other issues I could gather that this was not the intended behaviour.

**Comment by maniacx** [comment]
Check this settings.
![image](https://github.com/user-attachments/assets/35bbecdb-faf8-42ee-b1d9-36dcd2eeefec)

---

**With this settings disabled, it will use default Gnome behaviour.**
This extension doesnt control the icon, The icon is controlled by Gnome-shell

When battery is discharging, icon displayed is ![image](https://github.com/user-attachments/assets/5268aa43-3b10-4f51-a136-a60e2e605f8b)

When battery is in threshold (reported as pending-charge in upower), icon displayed is ![image](https://github.com/user-attachments/assets/5268aa43-3b10-4f51-a136-a60e2e605f8b)

When Battery is charging , icon displayed is ![image](https://github.com/user-attachments/assets/d9283b9c-e6d8-44b6-ac63-f6bd05ac7104)

---

**With this settings enabled, it will behave similar to Windows OS.**
The icon is controlled by the extension.

When battery is discharging, icon displayed is ![image](https://github.com/user-attachments/assets/5268aa43-3b10-4f51-a136-a60e2e605f8b)

When battery is in threshold (reported as pending-charge in upower), icon displayed is   ![image](https://github.com/user-attachments/assets/d9283b9c-e6d8-44b6-ac63-f6bd05ac7104)

When Battery is charging , icon displayed is ![image](https://github.com/user-attachments/assets/d9283b9c-e6d8-44b6-ac63-f6bd05ac7104)

---
To summarize, Gnome behavior, the icon reports more about battery status, whether charging, discharging or pending charge due to threshold.(while also reporting the battery level)

Well as the Windows,  the icon reports more about if charger is plugged-in or not (while also reporting the battery level)

For me, I want to know if my charger is (plugged with power) or (unplugged/ plugged but no-power)
So I introduce this feature.

Choose what kind of behavior of battery icon you want by enabling or disabling `Change battery indicator behavior`.
Let me know if you are seeing a different beahviour other than what I have described above.

**Comment by maniacx** [comment]
Closing, due to lack of feedback.
